lib/stm32/g0/rcc.c

```diff
@@ -0,0 +1,557 @@
+/** @defgroup rcc_file RCC
+ *
+ * @ingroup STM32G0xx
+ *
+ * @brief <b>libopencm3 STM32G0xx Reset and Clock Control</b>
+ *
+ * @author @htmlonly &copy; @endhtmlonly 2019 Guillaume Revaillot <g.revaillot@gmail.com>
+ *
+ * @date 10 January 2019
+ *
+ * This library supports the Reset and Clock Control System in the STM32 series
+ * of ARM Cortex Microcontrollers by ST Microelectronics.
+ *
+ * LGPL License Terms @ref lgpl_license
+ */
+
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**@{*/
+
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/pwr.h>
+#include <libopencm3/stm32/flash.h>
+#include <libopencm3/cm3/assert.h>
+
+/* Set the default clock frequencies after reset. */
+uint32_t rcc_ahb_frequency =  16000000;
+uint32_t rcc_apb1_frequency = 16000000;
+
+const struct rcc_clock_scale rcc_clock_config[RCC_CLOCK_CONFIG_END] = {
+	[RCC_CLOCK_CONFIG_LSI_32KHZ] = {
```

Comment:
how portable is thhis style? isn't it for creating sparse arrays?  I mean, it hhelps you find them, but... still.

Reply:
well, hm, I rechecked and this seem to be valid c99 (designated initializers) - Given the number of entries in that array, using them seemed safer.